### DISCUSSION TITLE
Mindstorms hub and keyboard key detection

### DISCUSSION THREAD

**Discussion by jeagh** [pybricks/support#892] [Q&A] [answered]
I'm having great success with pybricks and my Mindstorms hub and so I thank the developers for their efforts.

After successfully programming a colour sorter with pybricks I decided to try a remote control project. I have a simple wheeled vehicle that I'm controlling with code based on the keyboard example code (the 4WD example uses very similar code). I'm using the arrow keys for direction and any other key to stop.

However, if instead of pressing a key to stop the motor I'm wondering if I could simply release the forward key, for example, to stop the motor. I don't think it's possible to detect a released key but I thought I'd ask in case I've missed something.

From what I see on YouTube the Powered Up remote does allow a motor to stop just by releasing the button. I'll investigate this further.

**Comment by dlech** [comment, marked as answer]
The current method of using `stdin` doesn't actually detect key events, but rather the characters that would appear in a terminal as a result of typing. So if you keyboard is configured to automatically repeat characters when it is held down, you could detect the release of a key by the lack of repeating characters.

**Reply by jeagh** [reply]
Thank you yet again for answering my questions. I don't think my tablet Bluetooth keyboard is capable of repeating characters.

My original idea was to set a direction variable by detecting an arrow key press otherwise stop if no key was detected. I'll experiment further.
